base/rational.jl

```diff
@@ -10,23 +10,61 @@ struct Rational{T<:Integer} <: Real
     num::T
     den::T
 
-    function Rational{T}(num::Integer, den::Integer) where T<:Integer
-        num == den == zero(T) && __throw_rational_argerror_zero(T)
-        num2, den2 = divgcd(num, den)
-        if T<:Signed && signbit(den2)
-            den2 = -den2
-            signbit(den2) && __throw_rational_argerror_typemin(T)
-            num2 = -num2
+    function Rational{T}(num::Integer, den::Integer; checked::Bool) where T<:Integer
+        if checked
+            num == den == zero(T) && __throw_rational_argerror_zero(T)
```

Comment:
A minimal improvement of this line:
```
iszero(den) && iszero(num) && __throw
```
The differnece in the generated llvm code is:
before change
```
; │ @ /home/julia/julia/base/rational.jl:15 within `_#823'
; │┌ @ int.jl:444 within `==' @ promotion.jl:398
    %9 = icmp ne i64 %2, %3
; │└
   %10 = icmp ne i64 %3, 0
   %11 = or i1 %9, %10
   br i1 %11, label %L30, label %L28
```

after change
```
; │ @ /home/julia/julia/base/rational.jl:15 within `_#821'
   %9 = or i64 %2, %3
   %10 = icmp eq i64 %9, 0
   br i1 %10, label %L22, label %L25
```
- one `icmp` less than before.

Reply:
and using `iszero` `isempty` etc is best practice

Reply:
Done in the last commit